lib/saml_idp/xml_security.rb

```diff
@@ -77,6 +74,31 @@ def validate(idp_cert_fingerprint, soft = true, options = {})
         validate_doc(base64_cert, soft, options)
       end
 
+      def validate_with_sha256(idp_certificate, options = {})
+        if request_cert && Base64.decode64(request_cert) != idp_certificate.to_der
+          raise ValidationError.new('Request certificate not valid or registered', :request_cert_not_registered)
+        end
+
+        validate_doc(request_cert, false, options)
+      end
+
+      def request_cert
+        if cert_element.text.blank?
+          raise ValidationError.new(
+            'Certificate element present in response (ds:X509Certificate) but evaluating to nil',
+            :no_certificate_in_request
+          )
+        end
+
+        cert_element.text
+      end
+
+      def valid_cert(cert_text)
```

Comment:
WDYT of documenting the uncomment return value to help future us?
```suggestion
      # @return [Array(OpenSSL::X509::Certificate, false), Array(nil, Error)] tuple of (cert, error)
      def valid_cert(cert_text)
```